### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by alexander-yakushev**
I followed your suggestions in #337. I took some liberty regarding the Team page based on the number of commits/contributions and activity, please tell me if it should be changed.

### CODE REVIEW

**Comment by alexander-yakushev** [comment]
Sidenote: can you please trigger the build for 1.2 docs?
image

File: doc/modules/ROOT/pages/about/release_policy.adoc [hunk 1]
```diff
@@ -1,7 +1,6 @@
 = Release Policy
 
-We're following http://semver.org/[SemVer].footnote:[As much as one can be
-following it when the major version is 0.] At this point bumps of the
+We're following http://semver.org/[SemVer]. At this point bumps of the
```
Comment (bbatsov) [inline doc/modules/ROOT/pages/about/release_policy.adoc:3]: You can also remove much of what follows, as it's not really relevant right now (after the release of 1.0).

File: doc/modules/ROOT/pages/extensions.adoc [hunk 2]
```diff
@@ -58,24 +52,8 @@ hosted outside the nREPL GitHub organization.
...
-* https://github.com/technomancy/javert[Javert] provides a basic
-  object inspector. **(superseded by `cider-nrepl`)**
-* https://github.com/thunknyc/nrepl-profile[nrepl-profile]: profiling middleware,
-which was eventually integrated into `cider-nrepl`.
-
-
 [NOTE]
 ====
-This list doesn't aim to be complete. You can find more 3rd-party middleware listed
+More 3rd-party extensions (including deprecated/abandoned ones) are listed
```
Comment (bbatsov) [inline doc/modules/ROOT/pages/extensions.adoc:57]: Probably we can remove this section and the wiki page, as I had totally forgotten about them at this point. :D

File: doc/modules/ROOT/pages/troubleshooting.adoc [hunk 3]
```diff
@@ -43,42 +43,11 @@ If you encounter something like:
 nREPL server started on port 36457 on host 127.0.0.1 - nrepl://127.0.0.1:36457
 ----
 
-This means that you're trying to load middleware that's not targeting
-nREPL 0.4+. Make sure you're using the latest versions of all the
+This means that you're trying to load middleware incomatible with the current
```
Comment (bbatsov) [inline doc/modules/ROOT/pages/troubleshooting.adoc:46]: imcompatible

**Comment by bbatsov** [comment]
> Sidenote: can you please trigger the docs build for 1.2 tag?

I tried this a couple of times when I cut nREPL 1.2, but the build was failing and I forgot to investigate further. I'll try to figure this out today.

**Comment by bbatsov** [comment]
Oh, I'm an idiot - I used to use tags like `1.0.0` in nREPL for historical reasons, but all my other projects I used `v1.0.0` and by inertia I tagged 1.2 as `v1.2.0`, but wrote `1.2.0` in the Antora playbook... 🤦‍♂️

**Comment by alexander-yakushev** [comment]
Addressed the comments. Removing anything more from the Release Policy page will just make it say "We are using SemVer" and that's it:).

File: doc/modules/ROOT/pages/about/team.adoc [hunk 4]
```diff
@@ -34,3 +34,5 @@ initial design phases of nREPL:
 * Colin Jones
 * Laurent Petit
 * Eric Thorsen
+* Dominic Monroe
+* Christophe Grand
```
[the remark this answers is not in the document]
Comment (alexander-yakushev) [reply doc/modules/ROOT/pages/about/team.adoc:38]: My bad duplicating.

I see Reid has a single doc commit. Has he been doing something else as the core team?

[the remark this answers is not in the document]
Comment (bbatsov) [reply doc/modules/ROOT/pages/about/team.adoc]: Ah, I had forgotten about this - I think he was more action on cider-nrepl actually, where he worked on rich content types that we considered upstreaming to nREPL at some point, but the the experiment didn't get very far (just support for images in responses).

**Comment by bbatsov** [comment]
Changes deployed.

**Pull request merged**